docs/enhancements/backup-restore-support.md

```diff
@@ -0,0 +1,151 @@
+   * Remove relevant [code](https://github.com/openshift/assisted-service/blob/5bbf696fc0cc5a220fd8a0d1e408ab0bcc889e3f/internal/controller/controllers/agent_controller.go#L143-L151) from AgentController.
+
+2. Restore the Host record in DB according to properties from the Agent CR.
+   * Seems we can infer more than the mandatory properties:
+     * host.ID <-- agent.Name
+     * host.Role <-- agent.Role
+     * host.InfraEnvID <-- agent.metadata.annotations['infraenvs.agent-install.openshift.io']
+     * host.Kind <-- 'Host' or 'AddToExistingClusterHost' if cluster is already installed
+     * host.InstallerArgs <-- agent.InstallerArgs
+     * host.RequestedHostname <-- agent.Hostname
+   * Some of the inventory data can be restored from annotations.
+     E.g.:
+     * inventory.agent-install.openshift.io/cpu-architecture: x86_64
+     * inventory.agent-install.openshift.io/cpu-virtenabled: "true"
+     * inventory.agent-install.openshift.io/host-isvirtual: "true"
+     * inventory.agent-install.openshift.io/host-manufacturer: RedHat
+     * inventory.agent-install.openshift.io/host-productname: KVM
+     * inventory.agent-install.openshift.io/storage-hasnonrotationaldisk: "false"
+   * Note: We might need to add more annotations for additional inventory data.
+   
+### Risks and Mitigations
+
+* In general, these changes might affect flows on spoke cluster (before and after restore), need to ensure no regressions on all scenarios.
+* Adding the 'paused' annotation means a stale BMH status, need to check if it's crucial.
+* Keeping the Agent CR and restoring values into Host in DB could affect current flows (e.g. BindHost, UnbindHost, etc).
+* Restoring Hosts will result in partial records in DB. E.g. missing Inventory - could be challenging to tackle (I.e. probably need to add annotations to Agent).
+
+### Open Questions
+
+* Any specific concerns for the ConvergedFlow/Non-ConvergedFlow?
+* How seamlessly should we support restoring of 'in-progress' clusters (as the focus is on 'installed' clusters)?
+* To reduce risks, maybe worth adding an API (annotation) for manually pausing BMHs?
+
+### UI Impact
+
+* No UI changes are required.
+
+* The ACM UI should display the correct status for spoke clusters, etc.
+
+### Test Plan
+
+* All existing E2E tests should pass on restored/moved spoke clusters.
+* Should be tested both with ACM/OADP Backup/Restore feature and with the ZTP flow.
+
+## Current Alternatives
+
+* Using ACM/OADP, for including BMHs safely:
+  * Forcing BMHs in restore by adding an annotation: `cluster.open-cluster-management.io/backup: cluster-activation`
+  * Include BMHs original status during the restore (to avoid reprovisioning of the BMHs).
+    I.e. add 'restoreStatus' in the Restore CR
+    ```
+    apiVersion: cluster.open-cluster-management.io/v1beta1
+    kind: Restore
+    ...
+    spec:
+      restoreStatus:
+        includedResources:
+        - BareMetalHost
+    ```
```

Comment:
As a first pass/patch would replacing this with the BMH status and paused annotation be a net improvement even without all the rest? Seems like we could avoid this restore status stuff if we handle that internally.